src/manifest_reader.cpp

```diff
@@ -25,11 +25,13 @@ void ManifestReader::SetPartitionSpecID(int32_t partition_spec_id_p) {
 void ManifestReader::Initialize(unique_ptr<AvroScan> scan_p) {
 	const bool first_init = scan == nullptr;
 	scan = std::move(scan_p);
-	if (first_init) {
-		scan->InitializeChunk(chunk);
-	} else {
-		chunk.Reset();
+	if (!first_init) {
+		chunk.Destroy();
 	}
+	//! Reinitialize for every new scan, the schema isn't guaranteed to be the same for every scan
```

Comment:
Note: this came up because of the evolved partition spec.
The schema of the `partition` column of the `data_file` is affected by this.

This will likely proof problematic when we in the future try to use the MultiFileReader for the manifest reading, as the schemas can't be cleanly unified.